src/dist/replication/meta_server/meta_bulk_load_service.cpp

```diff
@@ -29,6 +30,232 @@ void bulk_load_service::initialize_bulk_load_service()
+}
+
+// ThreadPool: THREAD_POOL_META_STATE
+error_code bulk_load_service::check_bulk_load_request_params(const std::string &app_name,
+                                                             const std::string &cluster_name,
+                                                             const std::string &file_provider,
+                                                             const int32_t app_id,
+                                                             const int32_t partition_count,
+                                                             std::string &hint_msg)
+{
+    FAIL_POINT_INJECT_F("meta_check_bulk_load_request_params",
+                        [](dsn::string_view) -> error_code { return ERR_OK; });
+
+    // check file provider
+    dsn::dist::block_service::block_filesystem *blk_fs =
+        _meta_svc->get_block_service_manager().get_block_filesystem(file_provider);
+    if (blk_fs == nullptr) {
+        derror_f("invalid remote file provider type: {}", file_provider);
+        hint_msg = "invalid file_provider";
+        return ERR_INVALID_PARAMETERS;
+    }
+
+    // sync get bulk_load_info file_handler
+    const std::string remote_path = get_bulk_load_info_path(app_name, cluster_name);
+    dsn::dist::block_service::create_file_request cf_req;
+    cf_req.file_name = remote_path;
+    cf_req.ignore_metadata = true;
+    error_code err = ERR_OK;
+    dsn::dist::block_service::block_file_ptr file_handler = nullptr;
+    blk_fs
+        ->create_file(
+            cf_req,
+            TASK_CODE_EXEC_INLINED,
+            [&err, &file_handler](const dsn::dist::block_service::create_file_response &resp) {
+                err = resp.err;
+                file_handler = resp.file_handle;
+            })
+        ->wait();
+    if (err != ERR_OK || file_handler == nullptr) {
+        derror_f(
+            "failed to get file({}) handler on remote provider({})", remote_path, file_provider);
+        hint_msg = "file_provider error";
+        return ERR_FILE_OPERATION_FAILED;
+    }
+
+    // sync read bulk_load_info on file provider
+    dsn::dist::block_service::read_response r_resp;
+    file_handler
+        ->read(dsn::dist::block_service::read_request{0, -1},
+               TASK_CODE_EXEC_INLINED,
+               [&r_resp](const dsn::dist::block_service::read_response &resp) { r_resp = resp; })
+        ->wait();
+    if (r_resp.err != ERR_OK) {
+        derror_f("failed to read file({}) on remote provider({}), error = {}",
+                 file_provider,
+                 remote_path,
+                 r_resp.err.to_string());
+        hint_msg = "read bulk_load_info failed";
+        return r_resp.err;
+    }
+
+    bulk_load_info bl_info;
+    if (!::dsn::json::json_forwarder<bulk_load_info>::decode(r_resp.buffer, bl_info)) {
+        derror_f("file({}) is damaged on remote file provider({})", remote_path, file_provider);
+        hint_msg = "bulk_load_info damaged";
+        return ERR_INCOMPLETE_DATA;
```

Comment:
Would ERR_CORRUPTION code be more clear?

Reply:
In this case, will bulk load restart and finish normally?

Reply:
Yes, you are right,  `ERR_CORRUPTION` is better, I will update it.
In this case, `bulk_load_info` file can not be found on remote provider, bulk load process will not start.